RoboJackets/robocup-software
# Delete grSim and more launch file improvements

## Description
title. I still kept internal ref default for now becuase that is what erForce does too. Also allow us to switch sides in sim launch file. And other improvements (see changes to launch files). Can choose external ref or network radio instead of defaults by using ros2 launch syntax use_*:=False. See launch file for what the flags are called or makefile for examples

You can use that or if anyone wants to add more presets using this format to the makefile go ahead.

And finally, this pr fixes the sim2play race condition.

## Associated Issues
#1889 #1592

## Steps to test
Try a bunch of different launch setups to make sure it still works

Expected result: still works, now it is also no longer possible to use grSim.
